File: app/src/main/java/tech/cherri/tokenpushexample/JSONUtils.java
package tech.cherri.tokenpushexample;

public class JSONUtils {

    public static String pretty(String unformattedString) {
        StringBuilder stringBuilder = new StringBuilder();
        String indentString = "";

        for (int i = 0; i < unformattedString.length(); i++) {
            char letter = unformattedString.charAt(i);
            switch (letter) {
                case '{':
                case '[':
                    stringBuilder.append("\n" + indentString + letter + "\n");
                    indentString = indentString + "\t";
                    stringBuilder.append(indentString);
                    break;
                case '}':
                case ']':
                    indentString = indentString.replaceFirst("\t", "");
                    stringBuilder.append("\n" + indentString + letter);
                    break;
                case ',':
                    stringBuilder.append(letter + "\n" + indentString);
                    break;

                default:
                    stringBuilder.append(letter);
                    break;
            }
        }

        return stringBuilder.toString();
    }
}

File: kotlin/app/src/main/java/tech/cherri/tokenpushexample/JSONUtils.kt
package tech.cherri.tokenpushexample

import java.lang.StringBuilder

object JSONUtils {
    fun pretty(unformattedString: String): String {
        val stringBuilder = StringBuilder()
        var indentString = ""
        for (i in 0 until unformattedString.length) {
            val letter = unformattedString[i]
            when (letter) {
                '{', '[' -> {
                    stringBuilder.append(
                        """
    
    $indentString$letter
    
    """.trimIndent()
                    )
                    indentString = indentString + "\t"
                    stringBuilder.append(indentString)
                }
                '}', ']' -> {
                    indentString = indentString.replaceFirst("\t".toRegex(), "")
                    stringBuilder.append(
                        """
    
    $indentString$letter
    """.trimIndent()
                    )
                }
                ',' -> stringBuilder.append(
                    """
    $letter
    $indentString
    """.trimIndent()
                )
                else -> stringBuilder.append(letter)
            }
        }
        return stringBuilder.toString()
    }
}